syft/create_sbom.go

```diff
@@ -100,6 +100,11 @@ func setupContext(ctx context.Context, cfg *CreateSBOMConfig) (context.Context,
 }
 
 func setContextLicenseScanner(ctx context.Context, cfg *CreateSBOMConfig) (context.Context, error) {
+	// skip injecting a license scanner if one already set on context
+	if _, ok := ctx.Value(licenses.CtxKey).(licenses.Scanner); ok {
+		return ctx, nil
+	}
```

Comment:
What if we simply exported this function and changed the configuration from `*CreateSBOMConfig` to `cataloging.LicensesConfig`, which it looks like the only thing that is needed, below. You would be able to call this once and wouldn't need any further exported context management keys / etc.

Reply:
@kzantow Thank you - I like this idea too.

I've made the necessary changes, hopefully its in line with what you envisioned :)